OneCommunityGlobal/HighestGoodNetworkApp
# Jingyi-Interact with Task "Ready for Review" button

# Description
Give the user permission to interact with any "Ready for Review" task button to either mark it as complete or reset it with "More work needed, reset this button" 

## Related PRS (if any):
This frontend PR is related to the development backend PR.

## Main changes explained:
- use dispatch to check the permission for conditional rendering in React

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to "Other Links" -> "Permission Management" -> "Manage User Permissions"
6. In the permission management section, grant a volunteer account without the permission "interact with any 'Ready for Review'" permission.
7. log in to this volunteer account and in the dashboard -> tasks. Check if this volunteer can interact with the "Ready for Review" task button.

## Screenshots or videos of changes:
![Screenshot 2024-05-11 001835](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112589476/95a65d1a-df73-4c08-95d2-3aadd4787caf)

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112589476/b63d7713-57b3-4e93-9fe2-fa4c27c31ab0
